Gems/Atom/Feature/Common/Code/Source/PostProcessing/PaniniProjectionPass.cpp

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <PostProcessing/PaniniProjectionPass.h>
+#include <PostProcess/PostProcessFeatureProcessor.h>
+#include <Atom/RPI.Public/RenderPipeline.h>
+#include <Atom/RPI.Public/Scene.h>
+
+namespace AZ
+{
+    namespace Render
+    {
+        RPI::Ptr<PaniniProjectionPass> PaniniProjectionPass::Create(const RPI::PassDescriptor& descriptor)
+        {
+            RPI::Ptr<PaniniProjectionPass> pass = aznew PaniniProjectionPass(descriptor);
+            return AZStd::move(pass);
+        }
+
+        PaniniProjectionPass::PaniniProjectionPass(const RPI::PassDescriptor& descriptor)
+            : RPI::ComputePass(descriptor)
+        {
+        }
+
+        bool PaniniProjectionPass::IsEnabled() const
+        {
+            if (!ComputePass::IsEnabled())
+            {
+                return false;
+            }
+            const RPI::Scene* scene = GetScene();
+            if (!scene)
+            {
+                return false;
+            }
+            PostProcessFeatureProcessor* fp = scene->GetFeatureProcessor<PostProcessFeatureProcessor>();
+            const RPI::ViewPtr view = GetRenderPipeline()->GetDefaultView();
+            if (!fp)
+            {
+                return false;
+            }
+            PostProcessSettings* postProcessSettings = fp->GetLevelSettingsFromView(view);
+            if (!postProcessSettings)
+            {
+                return false;
+            }
+            const PaniniProjectionSettings* PaniniProjectionSettings = postProcessSettings->GetPaniniProjectionSettings();
+            if (!PaniniProjectionSettings)
+            {
+                return false;
+            }
+            return PaniniProjectionSettings->GetEnabled();
+        }
+
+        void PaniniProjectionPass::FrameBeginInternal(FramePrepareParams params)
+        {
+            // Must match the struct in PaniniProjection.azsl
+            struct Constants
+            {
+                AZStd::array<u32, 2> m_outputSize;
+                AZStd::array<float, 2> m_outputCenter;
+                float m_depth = PaniniProjection::DefaultDepth;
+            } constants{};
```

Comment:
Add a **AZStd::array<float, 3>m_pad** at the end to ensure that the struct ends at 16 byte boundary